### DISCUSSION TITLE
The problem with working on a local network

### DISCUSSION THREAD

**Discussion by CoderDodo2022** [open-webui/open-webui#1054] [General]
# Bug Report

## Description

**Bug Summary:**
[When you try to log in from another device connected to the same local network, you receive: "Account activation is waiting for an administrator to access the WebUI. The status of your account is currently awaiting activation. To access the WebUI, please contact the administrator. Administrators can manage user statuses from the admin panel."

**Steps to Reproduce:**
[I have set up an Ollamа Server Windows 10 computer for the snap-in, running on a local network and accepting requests on a subsequent request: http://192.168.0.219:11434 . 

On the server, I also installed a docker server, on which I created an administrator account, using this link: docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --open-webui name --always restart ghcr.io/open-webui/open-webui:main . 

On another device that I want to connect to (the device connects on the same LAN), I downloaded the docker container in accordance with the instruction: docker run -d -p 3000:8080 -e OLLAMA_API_BASE_URL=http://192.168.0.219:11434 -v open-webui:/app/backend/data --name is open-webui --restart is always ghcr.io/open-webui/open-webui:main]

**Expected Behavior:**
[I assumed that when trying to connect from another device, we would be able to send a request to the admin console on the main server so that we could confirm registration from the administrator account and use it safely. In this case, the account itself will be displayed on the main server.]

**Actual Behavior:**
[But in fact, when I log in from the second device, I only get a message about an incorrect username or password (if I try to enter administrator credentials), or when I try to register, I log in, but the message is displayed in full screen: "Account Activation Pending Contact Admin for WebUI Access Your account status is currently pending activation. To access the WebUI, please reach out to the administrator. Admins can manage user statuses from the Admin Panel." At the same time, no new accounts are displayed on the main server.]

## Environment

- **Windows 10 х64*
- **Yandex, Edge** 

## Reproduction Details

**Confirmation:**

- [ *] I have read and followed all the instructions provided in the README.md.
- [ *] I have reviewed the troubleshooting.md document.
- [ *] I have included the browser console logs.
- [ *] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

[Open webui via docker container, ollama server via .exe application ]

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by justinh-rahb** [comment]
Thank you for your detailed bug report. It seems there's a misunderstanding regarding how open-webui instances handle user databases. Each instance operates with its own local user database, which means separate instances won't automatically share user data or account statuses.

You actually don't need to install and run separate open-webui instances on each device. Instead, users on the same LAN can connect directly to the open-webui instance running on your server.

Based on the information you provided, here’s a simplified way to achieve what you're looking for:

1. **Setup open-webui on the Server Only**: Install and run the open-webui instance only on your server machine (e.g., the Windows 10 computer where you've set up the Ollama Server). Use the following command (which you've likely already done):
   ```bash
   docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main
   ```

2. **Accessing from Other Devices**: Instead of installing open-webui on each device, users on the same LAN can simply access the open-webui instance running on your server by navigating to `http://192.168.0.219:3000` (assuming `192.168.0.219` is the server's IP and `3000` is the port you've exposed). This way, all users interact with the same backend and user database, ensuring a consistent experience and shared data access.

3. **Ensure Network Accessibility**: Make sure your LAN settings and any firewalls are configured to allow traffic to and from the server on the required port (`3000` in this example).

This setup minimizes complexity and ensures all users on your LAN can access the same open-webui instance without needing separate installations. It also circumvents the issue you encountered with account activations not being shared across instances.

**Comment by CoderDodo2022** [comment]
Sorry for wasting your time, I'm new to docker and just starting to learn everything, thank you for your thorough and detailed help. 

Also, I will give the launch code of the docker container with which everything worked for me, for those who find themselves in the same situation as me: 

```docker run -d -p 192.168.0.219:3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main```

**Comment by justinh-rahb** [comment]
> Sorry for wasting your time, I'm new to docker and just starting to learn everything, thank you for your thorough and detailed help.

No waste of time at all, I commend you for following the template with a detailed description as it made forming an answer with an LLM very quick, thorough, and tailored to your scenario.

**Comment by zqrxiaer** [comment]
I also encountered this issue: "Account activation is waiting for an administrator to access the WebUI. The status of your account is currently awaiting activation. To access the WebUI, please contact the administrator. Administrators can manage user statuses from the admin panel."
I want to know how the administrator can activate the user's registered account. Thanks!

**Reply by justinh-rahb** [reply]
> I also encountered this issue: "Account activation is waiting for an administrator to access the WebUI. The status of your account is currently awaiting activation. To access the WebUI, please contact the administrator. Administrators can manage user statuses from the admin panel." I want to know how the administrator can activate the user's registered account. Thanks!

In the Admin section, when you click the "Pending" beside a user in the list, it will toggle them to "Active".

**Reply by zqrxiaer** [reply]
Thank you, I've solved my problem perfectly, thanks again for your answer!

**Comment by abidsoft** [comment]
I ran following command
docker run -d -p 192.168.0.219:3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main

it works well when i access it through IP on same PC (server)
but when i try to access it from another client through IP, it gives javascript error "Unexpected Token ?"

it can not load some of the dynamically loaded js files, maybe those javascrpt files are using local path instead of Absolute path

Help would be appreciated

Thanks

**Reply by abidsoft** [reply]
any update on this issue
